src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1481,8 +1481,54 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                                     value);
             return;
 
-        } else if (ASRUtils::is_complex(*left_type) && ASRUtils::is_complex(*right_type)) {
-            dest_type = left_type;
+        } else if (ASRUtils::is_complex(*left_type) || ASRUtils::is_complex(*right_type)) {
+
+            // handle cases when complex values are initialized in the form (3 + 4j) or (3j + 4)
+
+            // case: (complex j + real or integer)
+            if(ASRUtils::is_complex(*left_type) && (ASRUtils::is_integer(*right_type) || ASRUtils::is_real(*right_type))){
+                dest_type = left_type;
+                double right_value;
+
+                // right operand is real
+                if(ASRUtils::is_real(*right_type)){
+                    right_value = ASR::down_cast<ASR::ConstantReal_t>(
+                                                        ASRUtils::expr_value(right))->m_r;
```

Comment:
This would fail if the right hand side is a variable, not a number, correct?

Reply:
Yes you are correct. For it to work with variables, I think I need to insert a check to avoid throwing an exception when it is a variable. Then it will work at runtime for now. Later when we implement compile time value deduction for implicit casts, it will work in compile time.